### ISSUE TITLE
Form - required attribute missing for some of the components

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-react#1397]
All of the form inputs that can have the `required` attribute should be enabled to have it with the `isRequired` prop.

This attribute is supported on `<select>`, `<textarea>`, and several types of `<input>` ([Note on mdn about what `<input>` types don't support `required`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#attr-required))

**Comment by jgiardino** [comment]
On second review, it looks like textarea does receive the `required` attribute by using the `isRequired` prop, but not FormSelect. As part of this issue, please check that all form elements work as expected.

**Comment by jenny-s51** [comment]
Awaiting feedback from @jgiardino as to how to continue here

**Comment by jgiardino** [comment]
When I test the different components used within `<FormGroup>` I see that when I add `isRequired` to `<FormSelect>` that a `required` attribute is not added to the `<select>` in the html.

Here are the steps I used to produce this issue:
1. Navigate to this example: https://www.patternfly.org/v4/documentation/react/components/form/#form-horizontal
2. Click the Toggle Code button below the example.
3. Find the `<FormSelect>` component and add the prop `isRequired`
    - Note that the `isRequired` prop should also be added to the parent `<FormGroup>` whenever it's used on a form component. When added to `<FormGroup>`, the red asterisk will display in front of the label.
4. Open the inspector for the select input.

**Expected result:** the `required` attribute should be added to the `<select>` element
**Actual result:** the `required` attribute is missing

**Comment by jenny-s51** [comment]
per @tlabaj , this is not considered a breaking change as it adds an optional (but not required) prop
